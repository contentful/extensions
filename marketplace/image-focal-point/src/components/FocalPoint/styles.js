import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';
import hexRgb from 'hex-rgb';

const { red, green, blue } = hexRgb(tokens.colorElementMid);

export const BORDER_SIZE = 2;

export const styles = {
  focalPoint: css({
    backgroundColor: `rgba(${red}, ${green}, ${blue}, 0.5)`,
    borderRadius: '50%',
    border: `${BORDER_SIZE}px solid ${tokens.colorElementDarkest}`,
    opacity: 1,
    transition: `transform ${tokens.transitionDurationDefault} ${tokens.transitionEasingCubicBezier}`,
    position: 'absolute',
    pointerEvents: 'none'
  })
};
