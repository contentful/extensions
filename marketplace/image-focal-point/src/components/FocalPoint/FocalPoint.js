import React from 'react';
import PropTypes from 'prop-types';

import { BORDER_SIZE, styles } from './styles';

const SIZE = 32;

export const FocalPoint = ({ focalPoint }) => (
  <div
    style={{
      width: `${SIZE}px`,
      height: `${SIZE}px`,
      transform: `translate3d(${focalPoint.x - (SIZE / 2 + BORDER_SIZE)}px, ${focalPoint.y -
        (SIZE / 2 + BORDER_SIZE)}px, 0)`,
      top: 0,
      left: 0
    }}
    className={styles.focalPoint}
  />
);

FocalPoint.propTypes = {
  focalPoint: PropTypes.shape({
    x: PropTypes.number.isRequired,
    y: PropTypes.number.isRequired
  })
};
