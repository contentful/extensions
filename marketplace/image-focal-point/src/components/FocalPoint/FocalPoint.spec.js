import React from 'react';
import { render } from '@testing-library/react';

import { FocalPoint } from './FocalPoint';

const props = {
  focalPoint: {
    x: 10,
    y: 30
  }
};

describe('FocalPoint', () => {
  it('should render the focal point', () => {
    const { container } = render(<FocalPoint {...props} />);
    expect(container).toMatchSnapshot();
  });
});
