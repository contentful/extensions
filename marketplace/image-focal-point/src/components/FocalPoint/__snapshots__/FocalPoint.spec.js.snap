// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocalPoint should render the focal point 1`] = `
<div>
  <div
    class="css-hwcl17"
    style="width: 32px; height: 32px; transform: translate3d(-8px, 12px, 0); top: 0px; left: 0px;"
  />
</div>
`;
